src/patient-registration/before-save-prompt.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import Modal from 'carbon-components-react/es/components/Modal';
+import { useTranslation } from 'react-i18next';
+
+const getUrlWithoutPrefix = url => url.split(window['getOpenmrsSpaBase']())?.[1];
+const noop = () => {};
+
+interface BeforeSavePromptProps {
+  when: boolean;
+}
+
+const BeforeSavePrompt: React.FC<BeforeSavePromptProps> = ({ when }) => {
+  const history = useHistory();
+  const [open, setOpen] = useState(false);
+  const [newUrl, setNewUrl] = useState(undefined);
+  const { t } = useTranslation();
+
+  const cancelNavFn = useCallback(evt => {
+    if (!open && !evt.detail.navigationIsCanceled) {
+      evt.detail.cancelNavigation();
+      setNewUrl(evt.detail.newUrl);
+      setOpen(true);
+
+      // once the listener is run, we want to remove it immediately in case an infinite loop occurs due
+      // to constant redirects
+      evt.target.removeEventListener('single-spa:before-routing-event', cancelNavFn);
+    }
+  }, []);
```

Comment:
i need to add another dep here.

Reply:
also, currently getting an error tht the modal is triggered upon form submission - i need to pass the validation schema and validate the values of the form to bypass the modal.